src/sentry/api/endpoints/data_export_details.py

```diff
@@ -23,7 +23,7 @@ def get(self, request, organization, **kwargs):
 
         try:
             data_export = ExportedData.objects.get(id=kwargs["data_export_id"])
-            if request.GET.get("download") is not None:
+            if request.GET.get("download") is not None and data_export.file is not None:
```

Comment:
Even if a user sends a request with the download query parameter, it'll only work if there is a file attached to the ExportedData